coil-base/src/androidTest/java/coil/fetch/UriFetcherTest.kt

```diff
@@ -88,4 +87,35 @@ class UriFetcherTest {
         val result = loader.extractAssetFileName(uri)
         assertEquals(null, result)
     }
+
+    @Test
+    fun fileCacheKeyWithLastModified() {
+        val file = File(context.filesDir.absolutePath + File.separator + "file.jpg")
+        val uri = file.toUri()
+
+        // Copy the asset to filesDir.
+        val source = context.assets.open("normal.jpg").source().buffer()
+        val sink = file.sink().buffer()
+        sink.writeAll(source)
+
+        file.setLastModified(1234L)
+        val firstKey = loader.key(uri)
+
+        file.setLastModified(4321L)
+        val secondKey = loader.key(uri)
+
+        assertNotEquals(secondKey, firstKey)
+    }
+
+    @Test
+    fun assetCacheKeyWithLastModified() {
```

Comment:
I added a test for `file:///android_asset` Uris, as they always return 0 for `lastModified`.

Reply:
Thanks! I think this is good for the moment, but seems like this is a good point to separate `UriFetcher` and `AssetUriFetcher` since we don't need `File.lastModified` for an asset's cache key. I (or another community member) can address this in a separate up PR.

Reply:
I agree!  Honestly, you might also want a `ContentUriFetcher`, since it looks like the mime type resolution that is currently being used only works for `content://` Uris.  I ended up updating my custom one like so:

```kotlin
  @SuppressLint("DefaultLocale")
  private fun getMimeType(data: Uri): String? {
    return if (data.scheme == ContentResolver.SCHEME_CONTENT) {
      context.contentResolver.getType(data)
    } else {
      MimeTypeMap.getFileExtensionFromUrl(data.toString())?.let {
        val typeMap = MimeTypeMap.getSingleton()
        when {
          it == "image" -> "image/jpeg" // Handles some stupid iOS conventions.
          typeMap.hasMimeType(it) -> typeMap.getMimeTypeFromExtension(it.toLowerCase())
          else -> null
        }
      }
    }
  }
```

Reply:
Interesting, I'll circle back and take a look at this for 0.7.0.